subsys/power/power.c

```diff
@@ -18,6 +19,78 @@ LOG_MODULE_REGISTER(power);
 static int post_ops_done = 1;
 static enum power_states pm_state;
 
+#ifdef CONFIG_PM_CONTROL_OS_DEBUG
+
+struct pm_debug_info {
+	u32_t count;
+	u32_t last_res;
+	u32_t total_res;
+};
+
+static struct pm_debug_info pm_dbg_info[SYS_POWER_STATE_MAX];
+static u32_t timer_start, timer_end;
+
+#ifdef CONFIG_RTC
+static struct device *pm_timer;
+#endif
+
+static void sys_pm_debug_start_timer(void)
+{
+#ifdef CONFIG_RTC
+	if (!pm_timer) {
+		timer_start = rtc_read(pm_timer);
+	}
+#endif
+}
+
+static void sys_pm_debug_stop_timer(void)
+{
+#ifdef CONFIG_RTC
+	if (!pm_timer) {
+		timer_end = rtc_read(pm_timer);
+	}
+#endif
+}
+
+static void sys_pm_debug_setup_timers(void)
+{
+
+#ifdef CONFIG_RTC
+	pm_timer = device_get_binding(CONFIG_PM_DEBUG_RTC_DEV_NAME);
```

Comment:
Do we need dedicated RTC for that? Why we cannot use ticks or system timer cycles?

Reply:
agree. Not sure why we are using RTC here, not many platforms have this. Seems to be legacy from when power management was focused on devices with RTC driver (!= system timer)